great_expectations/experimental/datasources/interfaces.py

```diff
@@ -139,22 +146,27 @@ class DataAsset(ExperimentalBaseModel):
     # * type: Literal["csv"] = "csv"
     name: str
     type: str
-    order_by: List[BatchSorter] = Field(default_factory=list)
+
+    order_by_setter: BatchSortersDefinition = Field(
+        default_factory=list,
+        alias="order_by",
+    )
```

Comment:
I think I disagree or maybe just don't understand the use of `setter` fields and properties. It seems to complicate the code without an obvious benefit other than satisfying our internal type-checking. Is this something that's necessary as part of the `base_directory` change?

Reply:
This is the same as `regex`. Not required for this PR, but results in the removal of 10 `type: ignore`s (20 total). Another case of the types revealed in the IDE being different than what we actually pass to the model. It seems sloppy to just rely on `pydantic` to coerce types and sprinkle all these `ignore`s everywhere. They will continue to proliferate. In this case we actually use a `validator` to convert to `List[BatchSorter]`, but why don't we just pass the correct type to the model?

Reply:
Reverted this and handled it a different way. Let's discuss the `TypeAlias` `mypy` issue separately.